libcudacxx/include/cuda/std/__random/poisson_distribution.h

```diff
@@ -0,0 +1,347 @@
+        double __b1 = .4166667E-1 / __mean_;
+        double __b2 = .3 * __b1 * __b1;
+        __c3_       = .1428571 * __b1 * __b2;
+        __c2_       = __b2 - 15. * __c3_;
+        __c1_       = __b1 - 6. * __b2 + 45. * __c3_;
+        __c0_       = 1. - __b1 + 3. * __b2 - 15. * __c3_;
+        __c_        = .1069 / __mean_;
+      }
+    }
+
+    [[nodiscard]] _CCCL_API constexpr double mean() const noexcept
+    {
+      return __mean_;
+    }
+
+    [[nodiscard]] _CCCL_API friend constexpr bool operator==(const param_type& __x, const param_type& __y) noexcept
+    {
+      return __x.__mean_ == __y.__mean_;
+    }
+#if _CCCL_STD_VER <= 2017
+    [[nodiscard]] _CCCL_API friend constexpr bool operator!=(const param_type& __x, const param_type& __y) noexcept
+    {
+      return !(__x == __y);
+    }
+#endif // _CCCL_STD_VER <= 2017
+
+    friend class poisson_distribution;
+  };
+
+private:
+  param_type __p_{};
+
+  template <class _IntT,
+            class _FloatT,
+            bool _FloatBigger = (numeric_limits<_FloatT>::digits > numeric_limits<_IntT>::digits),
+            int _Bits         = (numeric_limits<_IntT>::digits - numeric_limits<_FloatT>::digits)>
+  [[nodiscard]] _CCCL_API static constexpr _IntT __max_representable_int_for_float() noexcept
+  {
+    static_assert(::cuda::std::is_floating_point<_FloatT>::value, "must be a floating point type");
+    static_assert(::cuda::std::is_integral<_IntT>::value, "must be an integral type");
+    static_assert(numeric_limits<_FloatT>::radix == 2, "FloatT has incorrect radix");
+    return _FloatBigger ? numeric_limits<_IntT>::max() : (numeric_limits<_IntT>::max() >> _Bits << _Bits);
+  }
+
+  template <class _IntT, class _RealT>
+  [[nodiscard]] _CCCL_API static _IntT __clamp_to_integral(_RealT __r) noexcept
+  {
+    using _Lim            = numeric_limits<_IntT>;
+    const _IntT __max_val = __max_representable_int_for_float<_IntT, _RealT>();
+    if (__r >= ::cuda::std::nextafter(static_cast<_RealT>(__max_val), INFINITY))
+    {
+      return _Lim::max();
+    }
+    else if (__r <= _Lim::lowest())
+    {
+      return _Lim::min();
+    }
+    return static_cast<_IntT>(__r);
+  }
+
+public:
+  // constructors and reset functions
+  _CCCL_API constexpr poisson_distribution() noexcept
+      : poisson_distribution{1.0}
+  {}
+  _CCCL_API constexpr explicit poisson_distribution(double __mean) noexcept
+      : __p_{__mean}
+  {}
+  _CCCL_API constexpr explicit poisson_distribution(const param_type& __p) noexcept
+      : __p_{__p}
+  {}
+  _CCCL_API constexpr void reset() noexcept {}
+
+  // generating functions
+  template <class _URng>
+  [[nodiscard]] _CCCL_API result_type operator()(_URng& __g)
+  {
+    return (*this)(__g, __p_);
+  }
+  template <class _URNG>
+  [[nodiscard]] _CCCL_API result_type operator()(_URNG& __urng, const param_type& __pr)
+  {
+    static_assert(__cccl_random_is_valid_urng<_URNG>, "");
+    double __tx;
+    uniform_real_distribution<double> __urd;
```

Comment:
While we cannot make this constexpr yet, I would love to already initialize those variables